cpp/include/cudf/groupby.hpp

```diff
@@ -73,6 +73,7 @@ struct aggregation_request {
 struct scan_request {
   column_view values;  ///< The elements to aggregate
   std::vector<std::unique_ptr<groupby_scan_aggregation>> aggregations;  ///< Desired aggregations
+  sorted values_pre_sorted{sorted::NO};  ///< Whether or not the values are sorted
```

Comment:
Why is this needed?

Reply:
I don't think this is correct. This could conflict with the `keys_are_sorted` information provided at construction of the `groupby` object. The keys of the groupby object and the values provided to an aggregation request must always be in the same order.

The optimization discussed in https://github.com/rapidsai/cudf/issues/8522 should only use information from `keys_are_sorted` passed at construction of `groupby`.